tangelo/toolboxes/molecular_computation/mm_charges_solver.py

```diff
@@ -0,0 +1,161 @@
+from tangelo.toolboxes.molecular_computation.molecule import atom_string_to_list
+
+
+class MMChargesSolver(abc.ABC):
+    """Instantiate electrostatic charges solver"""
+    def __init__(self):
+        pass
+
+    @abc.abstractmethod
+    def get_charges(self) -> List[Tuple[float, Tuple[float, float, float]]]:
+        """Obtain the charges for the given low parameters of the input fragment (e.g. filenames, list(s) of coordinates)
+
+        Returns:
+            List[float]: The list of charges
+            List[Tuple(float, Tuple(float, float, float))]: The atoms and geometries of the charges"""
+        pass
+
+
+def convert_files_to_pdbs(filenames: Union[str, List[str]]):
+    """Convert file or list of files to pdb files using openbabel.
+
+    Args:
+        filename (Union[str, List[str]]): The filename(s) that describe the MM region
+
+    Returns:
+        List[str]: The list of pdb equivalent filenames.
+    """
+
+    try:
+        import openbabel
+        from openbabel.openbabel import obErrorLog
+        obErrorLog.SetOutputLevel(0)
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(f"openbabel is required to use get_charges_and_coords_from_pdb_file when supplying only a pdb file."
+                                  "install with 'pip install openbabel-wheel'")
+
+    filesold = [filenames] if isinstance(filenames, str) else filenames
+
+    mol = openbabel.openbabel.OBMol()
+    conv = openbabel.openbabel.OBConversion()
+
+    files = []
+    for i, file in enumerate(filesold):
+        if file[-3:] != "pdb":
+            mol = openbabel.openbabel.OBMol()
+            conv.SetOutFormat("pdb")
+            conv.ReadFile(mol, file)
+            conv.WriteFile(mol, f"tmp{i}.pdb")
+            files += ["tmp{0}.pdb"]
+        else:
+            files += [file]
+
+    return files
+
+
+def get_default_mm_package() -> Type[MMChargesSolver]:
+    if is_package_installed("rdkit"):
+        return MMChargesSolverOpenMM()
+    elif is_package_installed("openmm"):
+        return MMChargesSolverRDKit()
+    else:
+        return None
+
+
+def get_mm_package(package: str = "default"):
+    """Return the MMCharges instance
+
+    Args:
+        package (str): The MM backend to use
+
+    Returns:
+        Type[MMChargesSolver]: The MMChargesSolver that computes the partial charges for given geometries
+    """
+    if package == "default":
+        return get_default_mm_package()
+    elif package.lower() == "openmm":
+        return MMChargesSolverOpenMM()
+    elif package.lower() == "rdkit":
+        return MMChargesSolverRDKit()
+    else:
+        raise ValueError(f"Requested MM package of {package} is not currently supported in Tangelo")
+
+
+class MMChargesSolverOpenMM(MMChargesSolver):
+    def __init__(self, force_field_params=('amber14-all.xml', 'amber14/tip3pfb.xml')):
+        self.force_field_params = force_field_params
+        pass
```

Comment:
pass means "do nothing" and is only useful if python complains that you forgot the body of a function or indentation does not work out.

Because of line 105, it is not needed here.